Fix pybind enum efficiency issue in return_and_correct_aliasing

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #161455
* #161432
* #161329
* #161328
* #161317
* __->__ #161315
* #161308
* #161304
* #161292
* #161301

Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

swolchok committed:
Fix pybind enum efficiency issue in return_and_correct_aliasing

Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

[ghstack-poisoned]

swolchok committed:
Update on "Fix pybind enum efficiency issue in return_and_correct_aliasing"


Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

[ghstack-poisoned]

swolchok committed:
Update on "Fix pybind enum efficiency issue in return_and_correct_aliasing"


Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

[ghstack-poisoned]

bdhirsh commented on torch/utils/_python_dispatch.py:
```diff
@@ -652,10 +660,16 @@ def get_alias_info(func) -> SchemaInfo:
         ]
-    schema_info = SchemaInfo(args=arg_schemas, outs=out_schemas)
+    schema_info = SchemaInfo(
+        args=arg_schemas, outs=out_schemas, int_tags=[int(x) for x in func.tags]
```
would it be reasonable to move this into a `SchemaInfo.__post_init__`, since this is just a perf optimization that can be inferred from the other args? Or are you worried about post_init itself being slow?

swolchok replied:
> are you worried about post_init itself being slow?

Not specifically, more that I had never heard of `__post_init__` before working on this stack.

I would rather execute fewer Python functions than more of them, though, and unless I'm wrong about SchemaInfo being confined to this file I'd rather not compromise performance for this one-callsite class.

swolchok committed:
rebase on "Fix pybind enum efficiency issue in return_and_correct_aliasing"


Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

[ghstack-poisoned]

swolchok committed:
Update on "Fix pybind enum efficiency issue in return_and_correct_aliasing"


Scanning a list of pybind enums with `in` is slow. See NOTE in code for full explanation.

This is a significant optimization; will be updating the torchdispatch/return_and_correct_aliasing portion of this stack with benchmark and results soonish.

[ghstack-poisoned]
